src/braket/default_simulator/density_matrix_simulation.py

```diff
@@ -157,8 +157,14 @@ def _apply_operations(
 
         for operation in operations:
             if isinstance(operation, (GateOperation, Observable)):
+                matrix = operation.matrix
                 result, temp = DensityMatrixSimulation._apply_gate(
-                    result, temp, qubit_count, operation.matrix, operation.targets, dispatcher
+                    result,
+                    temp,
+                    qubit_count,
+                    np.kron(matrix, matrix.conjugate()),
```

Comment:
Either move the `kron` call to inside `_apply_gate` or move the combined target construction here (preferably the former); otherwise, it can be confusing since it'll seem like the dimensions don't match in `_apply_gate`.